tasks/leave_restart_consul.yml

```
@@ -0,0 +1,19 @@
+---
+
+- name: Consul leave
+  delegate_to: "{{ consul_servers }}"
+  ansible.builtin.command: "{{ consul_binary }} leave {% if consul_acl_enable %} -token {{ consul_acl_master_token }} {% endif %} -http-addr {{ consul_addresses_http }}:{{ consul_ports_http }}"
+  changed_when: true
+
+- name: Restart consul on Unix
+  delegate_to: "{{ consul_servers }}"
```

```suggestion
  delegate_to: "{{ consul_server }}"
```

....if you agree with my corresponding suggestion in `handlers/restart_consul.yml`